fs/dafsclient/rmtclient.cpp

```diff
@@ -431,29 +436,28 @@ void receiveDaFsBuffer(ISocket * socket, MemoryBuffer & tgt, unsigned numtries,
             size32_t szread;
             SOCKREADTMS(socket)((gotLength<4000)?tgt.reserve(gotLength):tgt.reserveTruncate(gotLength), gotLength, gotLength, szread, timeout);
         }
-        catch (IJSOCK_Exception *e) {
-            if (e->errorCode()!=JSOCKERR_timeout_expired) {
-                EXCLOG(e,"receiveDaFsBuffer(1)");
+        catch (IException *e)
+        {
+            exception.setown(e);
+            tgt.setLength(origlen);
+        }
+
+        if (exception.get())
+        {
+            if (JSOCKERR_timeout_expired == exception->errorCode())
+            {
+                EXCLOG(exception,"receiveDaFsBuffer");
                 PrintStackReport();
-                if (!tm.timemon||!tm.timemon->timedout())
-                    flushDaFsSocket(socket);
             }
-            else {
-                EXCLOG(e,"receiveDaFsBuffer");
+            else
+            {
+                EXCLOG(exception,"receiveDaFsBuffer(1)");
                 PrintStackReport();
```

Comment:
I don't think there's any use/distinction between the if and else as far as the exception and stack logging is concerned, the (1) isn't useful, the exception text will already show that's it's a timeout expired exception.
So I'd move/common up the EXCLOG and PrintStackReport lines to above the if.

Reply:
ok, sure.